packages/mco/components/data-policies/create-dr-policy/create-dr-policy.tsx

```diff
@@ -138,31 +222,142 @@ type SyncScheduleProps = {
   setSchedule: React.Dispatch<React.SetStateAction<string>>;
 };
 
-export const CreateDRPolicy: React.FC<{}> = ({}) => {
+type ReRouteResourceProps = {
+  history: RouteComponentProps['history'];
+  match: Match<{ url: string }>;
+};
+
+type MirrorPeerMapping = {
+  [name in string]: MirrorPeerKind;
+};
+
+export const CreateDRPolicy: React.FC<ReRouteResourceProps> = ({match, history}) => {
   const { t } = useTranslation('plugin__odf-console');
-  const history = useHistory();
+  const { url }  = match;
+  
   const [policyName, setPolicyName] = React.useState('');
-  const [replication] = React.useState('');
-  const [selectedClusters, setSelectedClusters] = React.useState<Cluster[]>([]);
+  const [replication, setReplication] = React.useState('');
+  const [manageS3Profile, setManageS3Profile] = React.useState(false);
   const [syncTime, setSyncTime] = React.useState('5m');
+  const [isOCSDetected, setOCSDetected] = React.useState(false);
+  const [selectedClusters, setSelectedClusters] = React.useState<{[name in string] : Cluster}>({});
+  const [isReplicationOpen, setReplicationOpen] = React.useState(false);
+  const [isReplicationInputManual, setReplicationInputManual] = React.useState(false);
+  const [errorMessage, setError] = React.useState("");
+  const [mirrorPeerMapping, setMirrorPeerMapping] = React.useState<MirrorPeerMapping>({});
+
+
+  const [mirrorPeers, mirrorPeerLoaded, mirrorPeerLoadError] = useK8sWatchResource<MirrorPeerKind[]>({
+    kind: referenceForModel(MirrorPeerModel),
+    isList: true,
+    namespaced: false,
+  });
+
+  React.useEffect(() => {
+    setMirrorPeerMapping(mirrorPeers?.reduce((obj, mirrorPeer) => ({...obj, [mirrorPeer?.metadata?.name] : mirrorPeer}), {}));
+  }, [mirrorPeers, mirrorPeerLoaded, mirrorPeerLoadError]);
+
+  React.useEffect(() => {
+    if (Object.keys(selectedClusters)?.length === 2) {
```

Comment:
Store keys and values in a variable , we are requiring it often looks like
```suggestion
    if (Object.keys(selectedClusters ?? [])?.length === 2) {
```